### ISSUE TITLE
Module can be delayed to start loading.

### ISSUE THREAD

**Issue by BlishHUD** [blish-hud/Blish-HUD#109]
**Greaka at 7/19/2019 4:35:40 AM +00:00**
> BTW, my 3 marker packs took 10min to load yesterday
> Debug module active and debugger attached ^^
> Ah, and bhud has an unhandled exception when unloading the debug module

REF: https://discordapp.com/channels/531175899588984842/536970543736291346/601633233691607069

**Comment by dlamkins** [comment]
I can confirm that loading modules using tasks has not worked as expected.  The biggest problem is the amount needed to load some modules but how little time is yielded to them for them to finish their work.

I've found that the duration taken to load some modules exceeds `TaskCreationOptions.LongRunning` meaning tasks are not ideal for such durations.  Additionally, modules will often block other modules from loading due to not yielding enough between the module loading and the render loop of the core application.

You'll find that locking the FPS can change module load times as well as the core application is able to yield more frequently.

In tests, I've placed the `LoadAsync` call on its own thread and found much better performance all around.  Modules are loaded infrequently (when the application is started and when a module is enabled), so the extra overhead of a thread is no problem - even in cases where a module does not need to actually load much.
